packages/react-next/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react';
+import { classNamesFunction, mergeAriaAttributeValues, warnMutuallyExclusive, FocusRects } from '../../Utilities';
+import { Icon } from '../../Icon';
+import { ICheckboxProps, ICheckboxStyleProps, ICheckboxStyles } from './Checkbox.types';
+import { KeytipData } from '../../KeytipData';
+import { useId, useControllableValue } from '@uifabric/react-hooks';
+
+const getClassNames = classNamesFunction<ICheckboxStyleProps, ICheckboxStyles>();
+
+export const CheckboxBase = React.memo(
+  React.forwardRef((props: ICheckboxProps, ref: React.Ref<HTMLDivElement>) => {
+    const {
+      className,
+      disabled,
+      inputProps,
+      name,
+      boxSide = 'start',
+      theme,
+      ariaLabel,
+      ariaLabelledBy,
+      ariaDescribedBy,
+      styles,
+      checkmarkIconProps,
+      ariaPositionInSet,
+      ariaSetSize,
+      keytipProps,
+      title,
+      label,
+      onChange,
+    } = props;
+
+    const checkBox = React.useRef<HTMLInputElement>(null);
+    const [isChecked, setIsChecked] = useControllableValue(props.checked, props.defaultChecked, onChange);
+    const [isIndeterminate, setIsIndeterminate] = useControllableValue(props.indeterminate, props.defaultIndeterminate);
+
+    useDebugWarning(props);
+    useComponentRef(props, isChecked, isIndeterminate, checkBox);
+
+    const id = useId('checkbox-', props.id);
+    const classNames: { [key in keyof ICheckboxStyles]: string } = getClassNames(styles!, {
+      theme: theme!,
+      className,
+      disabled,
+      indeterminate: isIndeterminate,
+      checked: isChecked,
+      reversed: boxSide !== 'start',
+      isUsingCustomLabelRender: !!props.onRenderLabel,
+    });
+
+    const onRenderLabel = (): JSX.Element | null => {
+      return label ? (
+        <span aria-hidden="true" className={classNames.text} title={title}>
+          {label}
+        </span>
+      ) : null;
+    };
+
+    //#region Input callbacks
+    const _onFocus = React.useCallback(
```

Comment:
inline function like this is fine. no need to memoize (memoize should be done only if it's solving a real perf issue). checkout this article: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
same everywhere